prov/rxm/src/rxm_conn.c

```diff
@@ -657,9 +660,9 @@ rxm_process_connreq(struct rxm_ep *ep, struct rxm_eq_cm_entry *cm_entry)
 	ofi_addr_set_port(&peer_addr.sa, cm_entry->data.connect.port);
 
 	av = container_of(ep->util_ep.av, struct rxm_av, util_av);
-	peer = util_get_peer(av, &peer_addr);
-	if (!peer) {
-		RXM_WARN_ERR(FI_LOG_EP_CTRL, "util_get_peer", -FI_ENOMEM);
+	ret = util_get_peer(av, &peer_addr, &peer, 0);
+	if (ret) {
+		RXM_WARN_ERR(FI_LOG_EP_CTRL, "util_get_peer", ret);
 		goto reject;
 	}
```

Comment:
@j-xiong In my experiment, I just invoked `fi_av_insert()` to add the source address into av as follows:
```
ret = fi_av_insert(&av->util_av.av_fid, &peer_addr, 1, &peer->fi_addr, 0, NULL);
```
and then I'm planning to revise `xnet_cm_msg{}` to carry bind firewall information, and insert the address with `FI_FIREWALL_ADDR` if that is the case.

It seems to work well but I don't know if there is any side effect though

Reply:
I believe you are referring to a similar code segment in `xnet_rdm_cm.c`.
 
`struct xnet_rdm_cm` has a one-byte reserved field that can be used to carry a flag indicating if the connection request is coming from behind a firewall. The passive side can then pass `FI_FIREWALL_ADDR` flag to the `util_get_peer()` call here accordingly.  That will set the flag for the peer regardless of whether it has been inserted into the address vector or not.

Reply:
yeah - if there is no objection, I can start a prototype and create a PR for feedback. @shefty @ooststep FYI

Reply:
It sounds like a reasonable approach